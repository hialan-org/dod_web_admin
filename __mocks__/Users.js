export const users = [
    {
        userId: 1,
        email: 'test1@gmail.com',
        createdDate: 1588087949000,
        role: 'user',
        active: 1,
    },
    {
        userId: 2,
        email: 'test1@gmail.com',
        createdDate: 1588087949000,
        role: 'user',
        active: 1,
    }
]