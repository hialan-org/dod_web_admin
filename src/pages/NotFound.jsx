import React from 'react';
import {Link} from "react-router-dom";

function NotFound(props) {
    return (
        <div>
            <img src="https://i.imgur.com/6F707L7.jpg" title="source: imgur.com" style={{height: 600, display: 'block', margin: 'auto', position: 'relative' }}/>
        </div>
    )
}

export {NotFound}