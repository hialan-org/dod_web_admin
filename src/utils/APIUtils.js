import { API_BASE_URL, ACCESS_TOKEN } from '../constants/data';

const request = (options) => {
    const headers = new Headers({
        'Content-Type': 'application/json',
    })
    if(options.accessToken){
        headers.append('Authorization', 'Bearer ' + options.accessToken)
    } else {
        if(sessionStorage.getItem(ACCESS_TOKEN)) {
            headers.append('Authorization', 'Bearer ' + sessionStorage.getItem(ACCESS_TOKEN))
        }
    }

    const defaults = {headers: headers};
    options = Object.assign({}, defaults, options);

    return fetch(options.url, options)
        .then(response =>
            response.json().then(json => {
                if(!response.ok) {
                    return Promise.reject(json);
                }
                return json;
            })
        );
};

export function getCurrentUser() {
    if(!sessionStorage.getItem(ACCESS_TOKEN)) {
        return Promise.reject("No access token set.");
    }

    return request({
        url: API_BASE_URL + "/user/me",
        method: 'GET'
    });
}

export function login(accessToken) {
    return request({
        url: API_BASE_URL + "/loginWithGoogle",
        method: 'POST',
        accessToken: accessToken
    });
}

export function getListUser() {
    if(!sessionStorage.getItem(ACCESS_TOKEN)) {
        return Promise.reject("No access token set.");
    }
    return request({
        url: API_BASE_URL + "/users",
        method: 'GET'
    })
}

export function deleteUser(userId) {
    if(!sessionStorage.getItem(ACCESS_TOKEN)) {
        return Promise.reject("No access token set.");
    }
    return request({
        url: API_BASE_URL + "/users" + "/" + userId,
        method: 'DELETE',
    })
}

export function getTotalUserStock() {
    if(!sessionStorage.getItem(ACCESS_TOKEN)) {
        return Promise.reject("No access token set.");
    }
    return request({
        url: API_BASE_URL + "/statistics/countTotalUserStockNumber",
        method: 'GET'
    })
}

export function getTotalUserMoney() {
    if(!sessionStorage.getItem(ACCESS_TOKEN)) {
        return Promise.reject("No access token set.");
    }
    return request({
        url: API_BASE_URL + "/statistics/totalAmountOfUserMoney",
        method: 'GET'
    })
}