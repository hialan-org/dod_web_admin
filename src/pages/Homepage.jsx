import React, {useEffect, useState} from 'react';
import {EMAIL} from "../constants/data";
import {getListUser, login} from "../utils/APIUtils";

function Homepage(props) {
    const [email, setEmail] = useState("");

    useEffect(() => {
        if(props.authenticated){
            setEmail(sessionStorage.getItem(EMAIL))
        }
    }, []);

    return (
        <div style={styles.screen}>
            <div style={styles.body}>
                <h1>Dog of The Dow Management</h1>
                {
                    props.authenticated ?
                        (
                            <>
                                <h2 style={styles.text}>User: {email}</h2>
                                <a href="#" onClick={props.logout}>Logout</a>
                            </>
                        ) : ""
                }
            </div>

        </div>
    );
}

const styles = {
    screen : {
        display: "flex",
        height: '100vh',
        backgroundColor:"rgb(53,58,63)"
    },
    body : {
        flex:1,
        color: 'snow',

    }
};

export {Homepage};