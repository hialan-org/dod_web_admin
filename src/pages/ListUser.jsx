import React, {useState, useEffect} from "react";
import Table from "react-bootstrap/Table";
import {deleteUser, getListUser} from "../utils/APIUtils";
import {ROLE, ROLE_ADMIN, EMAIL} from "../constants/data";
import {Redirect} from "react-router-dom";

import 'react-bootstrap-table-next/dist/react-bootstrap-table2.min.css';
import BootstrapTable from 'react-bootstrap-table-next';
import overlayFactory from 'react-bootstrap-table2-overlay';
import Spinner from "react-bootstrap/Spinner";
import Button from "react-bootstrap/Button";
import Badge from "react-bootstrap/Badge";

function ListUser(props) {
    const [users, setUsers] = useState([]);
    const [redirect, setRedirect] = useState("");
    const [isLoading, setIsLoading] = useState(true);

    useEffect(() => {
        if ((sessionStorage.getItem(ROLE) && sessionStorage.getItem(ROLE) !== ROLE_ADMIN)
            || !props.authenticated) {
            setRedirect("/");
        } else {
            getListUser().then(result => {
                setUsers(result.body);
                setIsLoading(false);
            }).catch(err => {
                console.log(err);
            })
        }
    }, []);

    function onClickDelete(userId) {
        deleteUser(userId).then(result => {
            console.log("Delete success");
            getListUser().then(result => {
                setUsers(result.body);
                setIsLoading(false);
            }).catch(err => {
                console.log(err);
            })
        }).catch(err => {
            console.log(err);
        })
    }

    function buttonFormatter(cell, row, rowIndex, formatExtraData) {
        return (
            <Button
                disabled={row.email === sessionStorage.getItem(EMAIL)}
                variant="danger"
                onClick={() => onClickDelete(row.userId)}>
                Delete
            </Button>
        );
    }

    function dateFormatter(cell, row, rowIndex, formatExtraData) {
        const date = new Date(cell);
        return (
            <span>{date.toDateString()}</span>
        )
    }

    const columns = [{
        dataField: 'userId',
        text: 'User ID'
    }, {
        dataField: 'email',
        text: 'Email'
    }, {
        dataField: 'createdDate',
        text: 'Created Date',
        formatter: dateFormatter,
    }, {
        dataField: 'role',
        text: 'Role',
    }, {
        dataField: "actions",
        text: "Actions",
        sort: false,
        formatter: buttonFormatter,
        headerAttrs: {width: 100},
    }];

    return (
        redirect ?
            (<Redirect push to="/"/>)
            : (
                <>
                    <div className="list-users-header">
                            <Badge pill variant="primary">Total Users: {users.length}</Badge>
                            <Badge pill variant="secondary">Total Users's Stock: </Badge>
                            <Badge pill variant="secondary">Total Amount of User Money: </Badge>
                    </div>
                    <BootstrapTable
                        keyField='userId'
                        data={users}
                        columns={columns}
                        noDataIndication={() => <Spinner animation="border"/>}
                    />
                </>
            )
    )
}

export {ListUser}