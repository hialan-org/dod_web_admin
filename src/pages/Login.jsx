import React, {useState} from 'react';
import { Redirect } from "react-router-dom";
import {GoogleLoginButton} from "../components/GoogleLoginButton";
import {ACCESS_TOKEN, EMAIL, ROLE} from "../constants/data";
import {login} from "../utils/APIUtils";

function Login(props) {
    return (
        props.authenticated ?  (
            <Redirect push to="/"/>
        ) : (
            <div className="App">
                <GoogleLoginButton onClick={props.responseGoogle}/>
            </div>
        )
    );
}

export {Login}