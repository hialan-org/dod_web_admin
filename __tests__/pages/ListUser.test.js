import React from 'react';
import renderer from 'react-test-renderer';
import {ListUser} from "../../src/pages/ListUser";
import {users} from "../../__mocks__/Users";
import {getListUser} from "../../src/utils/APIUtils";
import BootstrapTable from 'react-bootstrap-table-next';

const {act} = renderer;

jest.mock('../../src/utils/APIUtils.js', () => ({
    getListUser: jest.fn()
}));

describe('<ListUser />', () => {
    it('if users table is not empty', async () => {
        getListUser.mockResolvedValue({
            body: users
        })
        let listUsers;
        await act(async () => {listUsers = renderer.create(
            <ListUser
                authenticated={true}
            />)
        });
        expect(listUsers.toJSON()).toMatchSnapshot();
        expect(listUsers.root.findByType(BootstrapTable).props.data.length)
            .toEqual(users.length);

    });
})