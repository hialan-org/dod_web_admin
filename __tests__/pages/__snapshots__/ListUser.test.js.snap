// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListUser /> if users table is not empty 1`] = `
Array [
  <div
    className="list-users-header"
  >
    <span
      className="badge badge-pill badge-primary"
    >
      Total Users: 
      2
    </span>
    <span
      className="badge badge-pill badge-primary"
    >
      Total Active Users: 
      2
    </span>
    <span
      className="badge badge-pill badge-secondary"
    >
      Total Users's Stock: 
    </span>
    <span
      className="badge badge-pill badge-secondary"
    >
      Total Amount of User Money: 
    </span>
  </div>,
  <div
    className="react-bootstrap-table"
  >
    <table
      className="table table-bordered"
    >
      <thead>
        <tr>
          <th
            tabIndex={0}
          >
            User ID
          </th>
          <th
            tabIndex={0}
          >
            Email
          </th>
          <th
            tabIndex={0}
          >
            Created Date
          </th>
          <th
            tabIndex={0}
          >
            Role
          </th>
          <th
            tabIndex={0}
          >
            Status
          </th>
          <th
            tabIndex={0}
            width={100}
          >
            Actions
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className={null}
          style={Object {}}
        >
          <td>
            1
          </td>
          <td>
            test1@gmail.com
          </td>
          <td>
            <span>
              Tue Apr 28 2020
            </span>
          </td>
          <td>
            user
          </td>
          <td>
            <span>
              Active
            </span>
          </td>
          <td>
            <button
              className="btn btn-danger"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Delete
            </button>
          </td>
        </tr>
        <tr
          className={null}
          style={Object {}}
        >
          <td>
            2
          </td>
          <td>
            test1@gmail.com
          </td>
          <td>
            <span>
              Tue Apr 28 2020
            </span>
          </td>
          <td>
            user
          </td>
          <td>
            <span>
              Active
            </span>
          </td>
          <td>
            <button
              className="btn btn-danger"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Delete
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
]
`;
